### ISSUE TITLE
flow_types breaks highlighting after ternary and bitshift

### ISSUE THREAD

**Issue by mischnic** [Thom1729/Sublime-JS-Custom#132]
### Sublime Text build number

4121

### Example Code

```TSX
const recordDefinition = (a) => {
    return a ? (a) : (a << 5);
};
d[105] = (data)=>{
    if (token >= 144 && token < 160 || token == 220 || token == 221) target = [];
    else target = {};
};
d[115] = ()=>new Set(read());
const typedArrays = ['BigUint64'].map((type)=>type + 'Array');
```

### JS Custom Preferences

```JSON with Comments
{
    "configurations": {
        "Default": {}
	},
	"defaults": {
		"file_extensions": ["js", "jsx"],
		"flow_types": true,
		"jsx": true
	}
}
```

### Configuration name

JS Custom - Default

### Description

The second line breaks the highlighting for the rest of the file.

It works correctly with `"flow_types": false`

![Bildschirmfoto 2021-11-21 um 13 30 02](https://user-images.githubusercontent.com/4586894/142761879-ea0a8773-cf4d-4aa9-9e51-db560db88ae3.png)

**Comment by Thom1729** [comment]
Minimal example:

```
return a ? (a) : (a < 5);
let x;
```

The bug is in Flow only, not in TypeScript, so there should hopefully be a simple fix.

**Comment by Thom1729** [comment]
This should be fixed in 4.3.0-beta.2.
